### ISSUE TITLE
Error when use ECDSA authentication with IKEv2

### ISSUE THREAD

**Issue by Sinkmanu** [libreswan/libreswan#283]
I am having a problem when try to connect via IKEv2 using ECDSA authentication.

I got the following error in logs:

```
Oct 28 17:32:05.433983: | verifying AUTH payload
Oct 28 17:32:05.433990: "ikev2"[1] 192.168.4.101 #1: authentication method: IKEv2_AUTH_ECDSA_P256 not supported in I2 Auth Payload
Oct 28 17:32:05.433999: "ikev2"[1] 192.168.4.101 #1: responding to IKE_AUTH message (ID 1) from 192.168.4.101:4500 with encrypted notification AUTHENTICATION_FAILED
```

I have generated the certificates with the following commands:
```
openssl ecparam -genkey -name secp256r1 -out CADH19.key
openssl req -x509 -new -SHA256 -subj '/CN=CA DH19' -nodes -key CADH19.key -days 3650 -out CADH19.crt

openssl ecparam -genkey -name secp256r1 -out serverDH19.key
openssl req  -new -SHA256 -key serverDH19.key -subj '/CN=192.168.4.10' -nodes -out serverDH19.csr
openssl req -in serverDH19.csr -noout -text
openssl x509 -req -SHA256 -extfile v3_server.ext -days 365 -in serverDH19.csr -CA CADH19.crt -CAkey CADH19.key -CAcreateserial -out serverDH19.crt

openssl ecparam -genkey -name secp256r1 -out clientDH19.key
openssl req  -new -SHA256 -key clientDH19.key -subj '/CN=192.168.4.101' -nodes -out clientDH19.csr
openssl req -in clientDH19.csr -noout -text
openssl x509 -req -SHA256 -extfile v3_client.ext -days 365 -in clientDH19.csr -CA CADH19.crt -CAkey CADH19.key -CAcreateserial -out clientDH19.crt
openssl pkcs12 -export -out clientDH19.pfx -inkey clientDH19.key -in clientDH19.crt
```
Also, the openssl template has _extendedKeyUsage = serverAuth,clientAuth,1.3.6.1.5.5.8.2.2_ for **IP Security IKE Intermediate**

After openssl commands i have imported these certs to _certutil_

 _ipsec.conf_ file:
```
config setup
  virtual-private=%v4:10.0.0.0/8,%v4:192.168.0.0/16,%v4:172.16.0.0/12,%v4:!192.168.42.0/24,%v4:!192.168.43.0/24
  protostack=netkey
  interfaces=%defaultroute
  uniqueids=no
  plutodebug=all
  plutostderrlog=/var/log/libreswan.log

conn ikev2
  left=%defaultroute
  leftcert=serverDH19
  leftid=@192.168.4.10
  leftsendcert=always
  leftsubnet=0.0.0.0/0
  right=%any
  rightid=%fromcert
  rightaddresspool=192.168.43.10-192.168.43.250
  rightca=%same
  authby=ecdsa-sha2_256
  narrowing=yes
  dpddelay=30
  dpdtimeout=120
  dpdaction=clear
  auto=add
  ikev2=yes
  rekey=no
  pfs=no
  ike-frag=no
  ike=aes128-sha2;modp2048,aes128-sha2;dh19,aes128-sha2;dh20,aes256-sha2;modp2048,aes128-sha1;modp2048,aes256-sha1;modp2048,aes256-sha2;dh19
  phase2alg=aes_gcm-null,aes128-sha1,aes256-sha1,aes128-sha2,aes256-sha2
  encapsulation=yes
  sha2-truncbug=yes
  modecfgdns="8.8.8.8 8.8.4.4"
```

Maybe I have some incorrect configuration in my ipsec.conf file. I have not found authyby=ecdsa in the manuals, just in the Wiki and in some links of libreswan. Someone knows whay could be happen?

**Comment by letoams** [comment]
We are aware of some issues with ECDSA usage, and I am working on a few fixes that will go into 3.30. Please check back in a few days or ping me in a few days and I should be able to give you a release candidate that is this all fixed.

Paul

**Comment by letoams** [comment]
this should work in the current release (3.32 or newer)
